src/Microsoft.AspNet.OData.Shared/Query/SkipTokenQueryOption.cs

```diff
@@ -0,0 +1,183 @@
+        /// Gets or sets the query setting 
+        /// </summary>
+        public ODataQuerySettings QuerySettings { get; private set; }
+
+        /// <summary>
+        /// Gets or sets the list of OrderByNodes for the original query
+        /// </summary>
+        public IList<OrderByNode> OrderByNodes { get; private set; }
+
+        /// <summary>
+        /// Apply the $skiptoken query to the given IQueryable.
+        /// </summary>
+        /// <param name="query">The original <see cref="IQueryable"/>.</param>
+        /// <param name="querySettings">The query settings to use while applying this query option.</param>
+        /// <param name="orderByNodes">Information about the orderby query option.</param>
+        /// <returns>The new <see cref="IQueryable"/> after the skiptoken query has been applied to.</returns>
+        public IQueryable<T> ApplyTo<T>(IQueryable<T> query, ODataQuerySettings querySettings, IList<OrderByNode> orderByNodes)
+        {
+            QuerySettings = querySettings;
+            OrderByNodes = orderByNodes;
+            return SkipTokenHandler.ApplyTo<T>(query, this) as IOrderedQueryable<T>;
+        }
+
+        /// <summary>
+        /// Apply the $skiptoken query to the given IQueryable.
+        /// </summary>
+        /// <param name="query">The original <see cref="IQueryable"/>.</param>
+        /// <param name="querySettings">The query settings to use while applying this query option.</param>
+        /// <param name="orderByNodes">Information about the orderby query option.</param>
+        /// <returns>The new <see cref="IQueryable"/> after the skiptoken query has been applied to.</returns>
+        public IQueryable ApplyTo(IQueryable query, ODataQuerySettings querySettings, IList<OrderByNode> orderByNodes)
+        {
+            QuerySettings = querySettings;
+            OrderByNodes = orderByNodes;
+            return SkipTokenHandler.ApplyTo(query, this);
+        }
+
+        /// <summary>
+        /// Validate the skiptoken query based on the given <paramref name="validationSettings"/>. It throws an ODataException if validation failed.
+        /// </summary>
+        /// <param name="validationSettings">The <see cref="ODataValidationSettings"/> instance which contains all the validation settings.</param>
+        public void Validate(ODataValidationSettings validationSettings)
+        {
+            if (validationSettings == null)
+            {
+                throw Error.ArgumentNull("validationSettings");
+            }
+
+            if (Validator != null)
+            {
+                Validator.Validate(this, validationSettings);
+            }
+        }
+
+        /// <summary>
+        /// Returns the handler for skip token
+        /// </summary>
+        /// <param name="context">The <see cref="ODataQueryContext"/> instance of the query.</param>
+        /// <returns></returns>
+        public static SkipTokenHandler GetSkipTokenHandler(ODataQueryContext context)
```

Comment:
>public static [](start = 8, length = 13)

The property itself is public set as:
public SkipTokenHandler SkipTokenHandler { get; set; }

So, do you still public static this method?

or verse vice?

Reply:
Made this to be internal and the property to be private. We should not have an instance to create a skiptoken handler. 

---
In reply to: [281401380](https://github.com/OData/WebApi/pull/1724#discussion_r281401380) [](ancestors = 281401380)